src/Fantomas.Core/MultipleDefineCombinations.fs

```diff
@@ -0,0 +1,284 @@
+
+    override this.Equals _ = false
+    override this.GetHashCode() = Int32.MinValue
+
+    interface IComparable with
+        member x.CompareTo other =
+            match other with
+            | :? CodeFragment as other -> (x :> IComparable<_>).CompareTo other
+            | _ -> -1
+
+    interface IComparable<CodeFragment> with
+        member x.CompareTo y =
+            match x, y with
+            // When comparing the different results of each format result, the single constant is that all hash lines
+            // should exactly match.
+            | CodeFragment.HashLine(line = lineX), CodeFragment.HashLine(line = lineY) ->
+                assert (lineX = lineY)
+                0
+            // Pick the other fragment is it has content you don't
+            | CodeFragment.NoContent _, CodeFragment.Content _ -> -1
+            // Pick our fragment is the other fragment has no code
+            | CodeFragment.Content _, CodeFragment.NoContent _ -> 1
+            // If both fragments are empty they are equivalent.
+            // Keep in mind that we could be comparing more the two fragments at the same time in `traverseFragments`
+            | CodeFragment.NoContent _, CodeFragment.NoContent _ -> 0
+            // If both fragments have content, we want to take the content with the most lines.
+            | CodeFragment.Content(lineCount = ownLineCount; code = ownContent),
+              CodeFragment.Content(lineCount = otherLineCount; code = otherContent) ->
+                let hasOwnContent = not (String.IsNullOrWhiteSpace ownContent)
+                let hasOtherContent = not (String.IsNullOrWhiteSpace otherContent)
+
+                if ownLineCount > otherLineCount then
+                    1
+                elif ownLineCount < otherLineCount then
+                    -1
+                elif
+                    String.IsNullOrWhiteSpace ownContent
+                    && not (String.IsNullOrWhiteSpace otherContent)
+                then
+                    -1
+                elif hasOwnContent && not hasOtherContent then
+                    1
+                elif not hasOwnContent && hasOtherContent then
+                    -1
+                else
+                    // This only really tailors to #1026
+                    // The shortest content is chosen because it will lead to the branch with less indentation.
+                    // This is again very specific to that exact unit test.
+                    let ownLength = ownContent.Length
+                    let otherLength = otherContent.Length
+
+                    if ownLength < otherLength then 1
+                    elif ownLength > otherLength then -1
+                    else 0
+            // This is an unexpected situation.
+            // You should never enter the case where you need to compare a hash line with something other than a hash line.
+            | x, other ->
+                // TODO: throw custom exception?
```

Comment:
Mmh, as of now I don't see that much of an advantage with a custom exception for this, so I wouldn't bother.